src/test/java/com/cgdpd/library/service/BookServiceTest.java

```diff
@@ -96,6 +110,103 @@ public void should_throw_exception_if_author_not_exist() {
     }
 
     @Test
+    void should_return_books_with_pagination() {
+        // given
+        var searchCriteria = SearchBookCriteria.builder().build();
+        Specification<BookEntity> spec = BookSpecifications.byBookSearchCriteria(searchCriteria);
+        int pageNumber = 0;
+        int pageSize = 10;
+        var sortParam = new SortParam("title", SortParam.Direction.ASC);
+        var paginationCriteria = new PaginationCriteria(pageNumber, pageSize,
+              Optional.of(sortParam));
+        var sort = sortParam.toDomainSort();
+        var pageable = PageRequest.of(pageNumber, pageSize, sort);
+
+        when(bookRepository.findAll(spec, pageable))
+              .thenReturn(new PageImpl<>(List.of(
+                    JOHN_DOE__THE_ADVENTUROUS__1987,
```

Comment:
Let's try to avoid using this data, it's purpose was for the functional tests, just create book from `aBookEntity().build()` modify something there if necessary